### PR TITLE
docs: update README with connection pooling

### LINKED ISSUE #195: Update README to recommend sqlalchemy implementation by default.
Update README to mention and recommend using sqlalchemy to manage connection pooling. It currently doesn't explain why in the sqlalchemy code sample.

### PR DESCRIPTION
**Pull request by jackwotherspoon**
Closes #195

### CODE REVIEW

**Review by enocom — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -94,6 +97,7 @@ engine = sqlalchemy.create_engine(
     creator=getconn,
 )
 ```
+**Note**: For more examples of using SQLAlchemy to manage connection pooling with the connector, please see [Cloud SQL SQLAlchemy Samples](https://github.com/GoogleCloudPlatform/python-docs-samples/tree/master/cloud-sql).
```
Comment (kurtisvg) [inline README.md:100]: Let's use this page instead https://cloud.google.com/sql/docs/mysql/manage-connections#python

File: README.md [hunk 2]
```diff
@@ -77,6 +77,9 @@ for row in result:
     print(row)
 ```
 
+#### Using a Connection Pool (RECOMMENDED)
```
Comment (kurtisvg) [inline README.md:80]: I'd like to see this part first. Ideally the person coming into the "How to use this connector" would receive a recommendation to use a connection pool within the first 2-3 sentences. 

I think this means we'd probably also remove the example without the pool, and show an example using the pool instead.

**Review by kurtisvg — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jackwotherspoon) [reply README.md]: How about this page as it shows the postgres version of the code sample? The link you recommended and the one I originally had don't show SQLAlchemy with the connector.
https://cloud.google.com/sql/docs/postgres/connect-connectors#python_1

File: README.md [hunk 3]
```diff
@@ -45,42 +45,20 @@ pip install git+https://github.com/GoogleCloudPlatform/cloud-sql-python-connecto
...
-for row in result:
-    print(row)
-```
+In the connector's `connect` method below, input your connection string as the first positional argument and the name of the database driver for the second positional argument. Insert the rest of your connection keyword arguments like user, password and database. You can also set the optional `timeout` or `ip_type` keyword arguments.
 
 To use this connector with SQLAlchemy, use the `creator` argument for `sqlalchemy.create_engine`:
 ```
-def getconn() -> pymysql.connections.Connection:
-    conn: pymysql.connections.Connection = connector.connect(
+def getconn():
```
Comment (kurtisvg) [inline README.md:60]: Why did we drop the typing?

**Review by kurtisvg — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (jackwotherspoon) [reply README.md:60]: I can add it back in if you want? I just thought it made the sample look a little cluttered. I'm open to adding it back in.

Comment (kurtisvg) [reply README.md:60]: I like typing. It's usually valuable if you're using typing, and easy to ignore if you aren't.

**Pull request merged**